styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -98,10 +98,9 @@ public QueuedStateManager(
   public CompletionStage<Void> trigger(WorkflowInstance workflowInstance, Trigger trigger) throws IsClosedException {
     ensureRunning();
 
-    final long nextCounter;
+    final long counter;
     try {
-      final long counter = storage.getLatestStoredCounter(workflowInstance).orElse(NO_EVENTS_PROCESSED);
-      nextCounter = counter + 1;
+      counter = storage.getLatestStoredCounter(workflowInstance).orElse(NO_EVENTS_PROCESSED);
```

Comment:
Looks like we lost a counter increment here. So a retrigger of a WFI will cause the last counter value to be reused.